### ISSUE TITLE
Container Lives after disconnect

### ISSUE THREAD

**Issue by willnx** [willnx/container_shell#11]
When you evoke the container_shell binary, and terminate the TTY, the container keeps on living.

This is problematic when leveraging ContainerShell via `ssh` if the user simply presses the `x` on the terminal instead of typing out `exit` or pressing `ctl c`. In that situation, you end with with a pile of running containers _just hanging around_, plus, the user's shell history wont ever be saved.

**Comment by willnx** [comment]
This looks like a problem within [dockerpty](https://github.com/d11wtq/dockerpty). 

This chunk of code doesn't appear (at first glance) to detect if the connection is gone:
https://github.com/d11wtq/dockerpty/blob/master/dockerpty/pty.py#L360

The last accepted merge to that source is from 2016, and there isn't much code to it.
If I'm right about the problem, getting a fix is likely going to be fork-lifting the code into this source.

**Comment by willnx** [comment]
Finally, progress!

After spending the day digging into this, I discovered that I don't have to touch dockerpty at all.
Turns out, OpenSSH will send [SIGHUP](http://man7.org/linux/man-pages/man7/signal.7.html) to the shell/process when it detects that the client has disconnected.

So if I add a [signal handler](https://docs.python.org/3/library/signal.html) to ContainerShell, I should be able to forward that message to the shell _inside_ the container. I think that would fix both problems because once the shell _inside_ the container terminates, the container does too. And bash seems to save history before gracefully terminating, so 🤞.

I know simply existing ContainerShell upon catching `SIGHUP` works to clean up the container, so at worst, we'll have that 😅
